api/envoy/api/v2/cds.proto

```diff
@@ -442,6 +442,15 @@ message Cluster {
   //   to an upstream host that becomes unhealthy, Envoy may spend a substantial amount of
   //   time exclusively closing these connections, and not processing any other traffic.
   bool close_connections_on_host_health_failure = 31;
+
+  // If this cluster uses EDS or STRICT_DNS to configure its host, immediately drain
```

Comment:
nit: s/host/hosts ? (This reads a bit rough to me, feel free to tweak in some other way if you think it would be better).